receiver/otlpreceiver/otlp.go

```diff
@@ -169,3 +173,40 @@ func (r *otlpReceiver) registerLogsConsumer(ctx context.Context, tc consumer.Log
 	}
 	return nil
 }
+
+// httpBodyCompressionHandler is a HTTP middleware checks the "Content-Encoding" HTTP header
+// and if a compression such as "gzip", "deflate", "zlib", is found,
+// the request body will be uncompressed accordingly.
+// It it encounters an error while do is, the original request is passed untouched.
+//
+// This was added to wrap the router generated by grpc-gateway as it does not provide a mechanism
+// to do this natively.
+func httpBodyCompressionHandler(handler http.Handler) http.Handler {
```

Comment:
I think this func is better to be in the "HttpServerSettings" so every receiver/exporter that uses this will share the behavior.

Reply:
Got it. This way we can also remove the duplicate code that's there in the zipkinreceiver module.